x-pack/plugins/maps/public/components/layer_panel/flyout_footer/index.js

```diff
@@ -20,17 +27,18 @@ const mapStateToProps = state => {
 
 const mapDispatchToProps = dispatch => {
   return {
-    cancelLayerPanel: () => {
-      dispatch(updateFlyout(FLYOUT_STATE.NONE));
-      dispatch(clearTemporaryStyles());
-      dispatch(clearTemporaryLayers());
+    cancelLayerPanel: async () => {
+      await dispatch(updateFlyout(FLYOUT_STATE.NONE));
+      await dispatch(clearTemporaryLayers());
+      await dispatch(rollbackToTrackedLayerStateForSelectedLayer());
     },
     saveLayerEdits: isNewLayer => {
       dispatch(updateFlyout(FLYOUT_STATE.NONE));
       dispatch(promoteTemporaryStyles());
```

Comment:
`promoteTemporaryStyles` should also be removed. All layer changes are made directly to the layer descriptor so nothing should need a temporary flag.

Reply:
:+1: